linkml/generators/markdowngen.py

```diff
@@ -31,15 +31,26 @@ class MarkdownGenerator(Generator):
     additionally, an index.md is generated that links everything together.
 
     The markdown is suitable for deployment as a MkDocs or Sphinx site
+
+    .. warning::
+        The MarkdownGenerator class is being deprecated in favor of DocGenerator which can
+        be found at the following path – `linkml/generators/docgen.py`. Going forward, DocGenerator
+        which can be invoked using the `gen-doc` command will be the preferred way to generate
+        Markdown documentation files for LinkML schemas.
     """
 
+    def __post_init__(self) -> None:
+        super().__post_init__()
```

Comment:
this would be where you want to put the call to `deprecation_warning`, but as is it doesn't seem like this does anything?

Reply:
Didn't fully realize that this was an alternative deprecation approach while I was implementing it, and you're exactly right, using the current deprecation approach with `deprecation_warning()` and making calls to it in `__post_init__()` and `cli()` should do the trick!